lib/Configuration.py

```diff
@@ -677,10 +680,21 @@ def TryGetNetworkFile(self, file=None, url=None, handler=None,
                         # Do I need to do something different for the progress handler?
                         retval.seek(0)
                         return retval
-                    log.error("Got http error %s" % str(error))
-                    url_exc = error
+                    elif error.response.status_code == HTTP_NOT_FOUND.value:
+                        # The requested file is not found on this server.
+                        url_exc = Exceptions.UpdateNetworkFileNotFoundException("Requested file %s not found" % (file if file else url))
+                        log.error("Error 404: %s" % str(url_exc))
+                    else:
+                        log.error("Got http error %s" % str(error))
+                        url_exc = Exceptions.UpdateNetworkServerException("Unable to load from url %s: " % (url, error.response.status_code))
+                        url_exc = error
+                except requests.exceptions.ConnectionError as e:
+                    log.error("Unable to connect to url %s: %s" % (url, str(e)))
+                    url_exc = Exceptions.UpdateNetworkConnectionException("Uable to connect to url %s" % url)
                 except BaseException as e:
+                    log.error("Unable to load %s, error class = %s" % (url, e.__class__))
                     log.error("Unable to load %s: %s", url, str(e))
+                    sys.exit(1)
```

Comment:
I dont think we should sys.exit here, is this a left over debug statement?

Reply:
Yes, thank you for catching it!